test/python/test_pack.py

```diff
@@ -725,3 +733,50 @@ def test_builddir_is_removed(cartridge_cmd, project_without_dependencies, pack_f
     assert process.returncode == 0
 
     assert not os.path.exists(builddir)
+
+
+@pytest.mark.parametrize('pack_format', ['rpm', 'deb'])
+def test_project_without_stateboard(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    STATEBOARD_ENTRYPOINT_NAME = 'stateboard.init.lua'
+
+    # remove stateboard entrypoint from project
+    os.remove(os.path.join(project.path, STATEBOARD_ENTRYPOINT_NAME))
+    project.distribution_files.remove(STATEBOARD_ENTRYPOINT_NAME)
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path,
+    ]
+
+    # call cartridge pack
+    process = subprocess.run(cmd, cwd=tmpdir)
+    assert process.returncode == 0
+
+    # packing should succeed with warning
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert "App directory doesn't contain stateboard entrypoint script" in output
+
+    # extract files from archive
+    archive_path = find_archive(tmpdir, project.name, pack_format)
+    extract_dir = os.path.join(tmpdir, 'extract')
+    os.makedirs(extract_dir)
+
+    if pack_format == 'rpm':
+        extract_rpm(archive_path, extract_dir)
+    elif pack_format == 'deb':
+        extract_deb(archive_path, extract_dir)
+        with tarfile.open(name=os.path.join(extract_dir, 'data.tar.xz')) as data_arch:
+            data_arch.extractall(path=extract_dir)
+
+    # check that stateboard unit file wasn't delivered
```

Comment:
Are there some tests where stateboard is delivered?

Reply:
Yes, see `check_systemd_dir` function in `utils.py`